src/main/java/glazer/paint/PencilTool.java

```diff
@@ -18,6 +19,7 @@ public void mousePressed(Graphics2D g, int x, int y,
 			BufferedImage image) {
 		// TODO Auto-generated method stub
 		g.setColor(properties.getColor());
+		g.setStroke(new BasicStroke(properties.getWeight()));
```

Comment:
Don't create a new Basic Stroke every single time the mouse is pressed. Move the basicstroke into PaintProperties and change this also on line 32